pkg/controller/registry/resolver/resolver.go

```diff
@@ -29,7 +29,7 @@ type constraintProvider interface {
 }
 
 type Resolver struct {
-	cache                     cache.OperatorCacheProvider
+	cache                     *cache.Cache
```

Comment:
Even if it were, I'd expect the interface to be defined by the consumer instead of exported from the cache package. I'm also not convinced that there will be a need to mock the cache. The contents of a cache are injectable, so tests are able to configure real objects that return what they want.

Reply:
> Even if it were, I'd expect the interface to be defined by the consumer instead of exported from the cache package. 

I share that perspective. It should probably be defined in the resolver package (or even a Cache interface with the method signatures the resolver cares about).

> The contents of a cache are injectable, so tests are able to configure real objects that return what they want.

My rule of thumb is to prefer abstracting with an interface (at the consumer, of course). I find that it helps to decouple the consumer tests from a particular implementation for two reasons:
- It's usually easier to generate specific call outcomes without wrestling a real implementation to do so; for example, errors/bad data, a specific sequence occurring with a non-deterministic operation (e.g. map-key iteration), etc
- Test-fixtures are usually smaller and more manageable